### ISSUE TITLE
gem install digest-sha3 fails on 2.5.2

### ISSUE THREAD

**Issue by tmornini** [rubinius/rubinius#3312]
```
v3 $ gem install digest-sha3
Building native extensions.  This could take a while...
ERROR:  Error installing digest-sha3:
	ERROR: Failed to build gem native extension.

    /Users/tmornini/.rbenv/versions/rbx-2.5.2/bin/rbx -r ./siteconf20150210-61855-sq41dt.rb extconf.rb
checking for ruby/digest.h... no
checking for rb_str_set_len()... no
creating Makefile

make "DESTDIR=" clean

make "DESTDIR="
compiling displayIntermediateValues.c
displayIntermediateValues.c:113:40: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
        fprintf(intermediateValueFile, text);
                                       ^~~~
1 warning generated.
compiling KeccakF-1600-reference.c
compiling KeccakNISTInterface.c
compiling KeccakSponge.c
compiling sha3.c
sha3.c:5:10: fatal error: 'digest.h' file not found
#include "digest.h"
         ^
1 error generated.
make: *** [sha3.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/tmornini/.rbenv/versions/rbx-2.5.2/gems/gems/digest-sha3-1.0.2 for inspection.
Results logged to /Users/tmornini/.rbenv/versions/rbx-2.5.2/gems/extensions/x86_64-darwin-14/2.1/digest-sha3-1.0.2/gem_make.out
```

**Comment by brixen** [comment]
@tmornini the digest.h file is part of the digest C-ext. Apparently, MRI copies it to its own include directory at some point. We'll probably need to do something similar with the pre-installed rubysl-digest gem.

Is there a Ruby alternative to this gem?

**Comment by tmornini** [comment]
@brixen Yes, there are ruby alternatives. This one was #1 on Google for me though. :-)

**Comment by bjfish** [comment]
@brixen I think this happens around line 2317   https://github.com/ruby/ruby/blob/af5b7053f17b61681b7c492800cc8bf347de1cdd/lib/mkmf.rb

**Comment by brixen** [comment]
@tmornini this is the first time we've ever hit this issue. I wonder if a good solution would be to just include sha3 into Digest rubysl-digest gem. There's probably not a good reason it's not included in MRI other than MRI bundled standard libraries often being out of date.

**Comment by tmornini** [comment]
@brixen Shouldn't the gem I used just work? Happy to open a PR on the gem if it's something they need to fix.

**Comment by brixen** [comment]
@tmornini the question isn't whether it should work on not, but what we should do to make it work. The situation is somewhat like a Ruby library that expects to use the Ruby Digest class, but without calling `require "digest"` (which it should, because Digest is part of the standard library, which is available via `require`) because it can just go look at the path where the the file is installed.

In this case, an external C-ext depends on a file that is part of a standard library gem. There's no equivalent C mechanism for `require "digest"`, where here it would pull in the header file that is part of Digest.

Ideally, this digesh-sha3 wouldn't be a separate library. The standard library Digest should include it. So, including it in rubysl-digest is one option for us.

Another option is to copy all header files from all standard library C-ext gems into the Rubinius include directory. I'm not completely opposed to it but it's nasty. If this is the only gem we've hit that expects to be able to access another library's header files, maybe we shouldn't support this.

**Comment by tmornini** [comment]
@brixen I see.

Am I correct in my belief that if the gem had a ```'require 'digest'``` in the code, all would be fine?

**Comment by brixen** [comment]
@tmornini no, `require 'digest'` wouldn't help here. The digest-sha3 C-ext is trying to use a header file that is part of digest C-ext.

**Comment by tmornini** [comment]
@brixen Got it.

**Comment by brixen** [comment]
@tmornini the fixed version of rubysl-digest (2.0.8) will be pre-installed in the next Rubinius release, but if you need it in 2.5.5, you can install it like this:

```
gem install -i `rbx -e 'puts Rubinius::GEMS_PATH'` rubysl-digest -v 2.0.8
```
